src/ArrayWriter.cpp

```diff
@@ -463,7 +463,16 @@ template<>
 void PopulateBuilder<arrow::Type::INT16>(shared_ptr<arrow::DataType> datatype, K k_array, arrow::ArrayBuilder* builder, TypeMappingOverride& type_overrides)
 {
   auto int16_builder = static_cast<arrow::Int16Builder*>(builder);
-  PARQUET_THROW_NOT_OK(int16_builder->AppendValues((int16_t*)kH(k_array), k_array->n));
+  if( type_overrides.null_mapping.have_int16 ){
+    std::vector<bool> null_bitmap( k_array->n );
+    for( auto i = 0ll; i < k_array->n; ++i ){
+      null_bitmap[i] = type_overrides.null_mapping.int16_null ^ kH( k_array )[i];

```

Comment:
Does that not need to be:
```
null_bitmap[i] = !(type_overrides.null_mapping.int16_null ^ kH( k_array )[i]);
```
Or actually, I think I was overcomplicating thing by suggesting an xor.  This would be fine:
```
null_bitmap[i] = type_overrides.null_mapping.int16_null == kH( k_array )[i];
```
since it's a simple compare, not a branch

Reply:
My mistake `^` if fine (or `!=` may be more readable)